### PR TITLE
feat: add AWS Secret Manager feature to store DSN

### PR DESCRIPTION
**Pull request by elturkym**
This change allow AWS users, to store the database credentials  in AWS Secret manager instead of adding the credentials in the configuration file directly. 

Changes:
- Adding a new config `aws_secret_name` under `target` to store the data source name (DSN) which has the database connection info. 
- Update the config.go to read the DSN from the given secret in the configuration file.
- Update the config.go to override the DSN from AWS secret manager. 
- Update README file with the steps to use AWS Secret manager.

### CODE REVIEW

**Comment by burningalchemist** [comment]
Hey @elturkym, thank you for your contribution!

Could you please fix the comment style (coming from `gofmt` in the build log)? Apart from that, looks good to me! 👍

File: config/config.go [hunk 1]
```diff
@@ -239,6 +250,47 @@ func (t *TargetConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return checkOverflow(t.XXX, "target")
 }
 
+//
+//AWS Secret
+//
```
Comment (burningalchemist) [inline config/config.go:255]: As per the `gofmt -d` output:

```suggestion
//AWS Secret
```

Comment (burningalchemist) [reply config/config.go]: I think it's still not enough. We also need to remove heading and trailing slashes. Just like in the suggestion I made above. 🙂👍

**Comment by elturkym** [comment]
My apologies, looks like there are new formatting issues, I will look at them.

File: config/config.go (hunk 1 above)
Comment (elturkym) [reply config/config.go]: Thanks, will address that

**Comment by burningalchemist** [comment]
Sure, no worries. Check my last comment. 🙂👍

**Comment by burningalchemist** [comment]
Hey @elturkym, do you need any help or assistance from my end to address the remaining issues? :)

**Comment by elturkym** [comment]
I was trying to run the workflow locally before submitting a new PR, but let me raise a new PR and check if there is still an error

**Review by burningalchemist — APPROVED** [review]
LGTM! Thanks! 🚀

**Pull request merged**